components/api_server/src/routes/source.py

```diff
@@ -200,16 +200,16 @@ def _source_description(data_model, items, edit_scope, parameter_key, old_value)
     return source_description
 
 
-def _availability_checks(data_model, source, parameter_key: str) -> list[dict[str, str | int]]:
+def _availability_checks(source, parameter_key: str) -> list[dict[str, str | int]]:
     """Check the availability of the URLs."""
-    source_model = data_model["sources"][source["type"]]
-    parameters = source_model["parameters"]
-    token_validation_path = parameters.get("private_token", {}).get("validation_path", "")
+    parameters = DATA_MODEL.sources[source["type"]].parameters
+    private_token = cast(PrivateToken, parameters.get("private_token"))
+    token_validation_path = private_token.validation_path if private_token else ""
```

Comment:
Too bad, it seems that casting None circumvents the `__init__`. Otherwise, an empty object would've been nice :(

Reply:
I guess the ideal solution would if `parameters` was a typed dict so mypy would know that the values of the key `private_token` are of type `PrivateToken`. But I'm not going to touch the data model before we've upgraded to Pydantic 2.